.github/templates/check_src_changes/action.yml

```diff
@@ -0,0 +1,93 @@
+name: Check source file changes
+
+outputs:
+  modified_modules:
+    description: "Space deliminated list of modified modules"
+    value: ${{ steps.output-changes.outputs.modified_modules }}
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Find changed files inside action folder
+      id: changed-files-action
+      uses: tj-actions/changed-files@v42
+      with:
+        files: src/action/**
+      
+    - name: Get changed files inside interfacing folder
+      id: changed-files-interfacing
+      uses: tj-actions/changed-files@v42
+      with:
+        files: src/interfacing/**
+    
+    - name: Get changed files inside perception folder
+      id: changed-files-perception
+      uses: tj-actions/changed-files@v42
+      with:
+        files: src/perception/**
+    
+    - name: Get changed files inside samples folder
+      id: changed-files-samples
+      uses: tj-actions/changed-files@v42
+      with:
+        files: src/samples/**
+    
+    - name: Get changed files inside simulation folder
+      id: changed-files-simulation
+      uses: tj-actions/changed-files@v42
+      with:
+        files: src/simulation/**
+    
+    - name: Get changed files inside world_modeling folder
+      id: changed-files-world-modeling
+      uses: tj-actions/changed-files@v42
+      with:
+        files: src/world_modeling/**
+
+    - name: Create list of changed modules
+      env:
+        CHANGED_MODULES: ""
+      run: |
+        if [ ${{ steps.changed-files-action.outputs.any_changed }} == 'true' ]; then
+          echo "Detected action changes"
+          CHANGED_MODULES+="action "
+        fi
+
+        if [ ${{ steps.changed-files-interfacing.outputs.any_changed }} == 'true' ]; then
+          echo "Detected interfacing changes"
+          CHANGED_MODULES+="interfacing "
+        fi
+
+        if [ ${{ steps.changed-files-perception.outputs.any_changed }} == 'true' ]; then
+          echo "Detected perception changes"
+          CHANGED_MODULES+="perception "
+        fi
+
+        if [ ${{ steps.changed-files-samples.outputs.any_changed }} == 'true' ]; then
+          echo "Detected samples changes"
+          CHANGED_MODULES+="samples "
+        fi
+
+        if [ ${{ steps.changed-files-simulation.outputs.any_changed }} == 'true' ]; then
+          echo "Detected simulation changes"
+          CHANGED_MODULES+="simulation "
+        fi
+
+        if [ ${{ steps.changed-files-world-modeling.outputs.any_changed }} == 'true' ]; then
+          echo "Detected world_modeling changes"
+          CHANGED_MODULES+="world_modeling"
+        fi
+
+        echo "CHANGED_MODULES=$CHANGED_MODULES" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Output list of changed modules
+      id: output-changes
+      run: |
+        echo "The changed modules are: $CHANGED_MODULES"
+        echo "modified_modules=$CHANGED_MODULES" >> $GITHUB_OUTPUT
```

Comment:
could you put this a `check_src_changes.sh` file? some similar to `github_context.sh`

Reply:
you can also combine these two bash scripts together, just section it off with comments if u want

Reply:
Yep sounds good, I put both of them into one bash script. I had to pass the `$ {{ steps.changed-___.outputs.any_changed}}` as environment variables though.